### PR TITLE
Follow/unfollow

### PR DESCRIPTION
**Pull request by chanchal16**
- Added follow/unfollow functionality
- user can follow a person
- user can unfollow a person
- user can see followers/following list

### CODE REVIEW

File: src/features/profile/profileSlice.js [hunk 1]
```diff
@@ -43,14 +40,29 @@ export const getUserPost = createAsyncThunk(
...
+		return data
+	}catch(err){console.error(err)}
+})
+// unfollow user
+export const unFollowUser = createAsyncThunk("users/unFollowUser",async({token,userId})=>{
+	try{
+		const {data} = await axios.post(`/api/users/unfollow/${userId}`,{},
+		{headers:{authorization:token}})
+		return data
+	}catch(err){console.error(err)}
```
Comment (ajay-mandaviya) [inline src/features/profile/profileSlice.js:64]: try to make api  call function  in another function

File: src/features/profile/UserProfileInfo.jsx [hunk 2]
```diff
@@ -1,20 +1,50 @@
...
+import { useParams} from "react-router-dom";
 
 export const UserProfileInfo = ({userData}) => {
+    const {user} = useSelector(state => state.auth)
+    const{token} = user
+    const {username} = useParams();
+    const dispatch = useDispatch()
+    // check if following 
+    const isFollowing = userData?.followers.some(user=>user.username !== username)
+
```
Comment (ajay-mandaviya) [inline src/features/profile/UserProfileInfo.jsx:14]: make a  common some function  i think you need it  multiple time

**Review by ajay-mandaviya — APPROVED** [review]

File: src/features/profile/profileSlice.js (hunk 1 above)
Comment (chanchal16) [reply src/features/profile/profileSlice.js:64]: Will refactor it later.

File: src/features/profile/UserProfileInfo.jsx (hunk 2 above)
Comment (chanchal16) [reply src/features/profile/UserProfileInfo.jsx:14]: No this condition is needed only for this feature. I have made functions for other conditions.

File: src/features/profile/profileSlice.js (hunk 1 above)
Comment (ajay-mandaviya) [reply src/features/profile/profileSlice.js:64]: nothing wrong with this

**Review by Sharmaryan — APPROVED** [review]

File: src/features/profile/profileSlice.js (hunk 1 above)
Comment (chanchal16) [reply src/features/profile/profileSlice.js:64]: ok, will refactor it later.

**Pull request merged**